client/src/pages/lessons/edit.tsx

```diff
@@ -0,0 +1,579 @@
+  palette: {
+    primary: {
+      main: "#1b6a9c",
+    },
+  },
+});
+
+const useStyles = makeStyles({
+  root: {
+    width: "100%",
+    "& .MuiTextField-root": {
+      margin: theme.spacing(1),
+      marginLeft: theme.spacing(1),
+      marginRight: theme.spacing(1),
+      width: "100ch",
+    },
+    "& > *": {
+      borderBottom: "unset",
+    },
+  },
+  button: {
+    margin: theme.spacing(2),
+  },
+  heading: {
+    fontSize: theme.typography.pxToRem(15),
+    flexBasis: "33.33%",
+    flexShrink: 0,
+  },
+  secondaryHeading: {
+    fontSize: theme.typography.pxToRem(15),
+    color: theme.palette.text.secondary,
+  },
+  container: {
+    maxHeight: 440,
+  },
+});
+
+type ExpectationProp = {
+  row: LessonExpectation;
+  index: number;
+};
+
+const LessonEdit = ({ search }: { search: any }) => {
+  let { lessonId } = search;
+  const originalId = lessonId;
+  //console.log(lessonId);
+  const classes = useStyles();
+  const inititialLesson = {
+    lessonId: "",
+    name: "Lesson Name",
+    intro: "Introduction",
+    question: "Question",
+    conclusion: ["Add a clossing to the lesson"],
+    expectations: [
+      {
+        expectation: "Add ideal answer for an expectation",
+        hints: [{ text: "Add a hint to help for the expectaion" }],
+      },
+    ],
+  };
+
+  const [lesson, setLesson] = React.useState(inititialLesson);
+  const [change, setChange] = React.useState(false);
+  const [create, setCreate] = React.useState(false);
+  const [expectationOpen, setExpectationOpen] = React.useState(true);
+
+  React.useEffect(() => {
+    if (lessonId !== "new") {
+      let mounted = true;
+      fetchLesson(lessonId)
+        .then((lesson: Lesson) => {
+          console.log("fetchLesson got", lesson);
+          if (mounted) {
+            if (lesson !== undefined) {
+              setLesson(lesson);
+            }
+          }
+        })
+        .catch((err: any) => console.error(err));
+      return () => {
+        mounted = false;
+      };
+    } else {
+      lessonId = uuid();
+      console.log(lessonId);
+      setLesson({ ...lesson, lessonId: lessonId });
+    }
+  }, []);
+
+  function handleLessonNameChange(name: string): void {
+    setChange(true);
+    setLesson({ ...lesson, name: name });
+  }
+
+  function handleLessonIdChange(lessonId: string): void {
+    setChange(true);
+    setLesson({ ...lesson, lessonId: lessonId });
+  }
+
+  function handleIntroChange(intro: string): void {
+    setChange(true);
+    setLesson({ ...lesson, intro: intro });
+  }
+
+  function handleQuestionChange(question: string): void {
+    setChange(true);
+    setLesson({ ...lesson, question: question });
+  }
+
+  function handleConclusionChange(con: string, index: number): void {
+    setChange(true);
+    const copyLesson = { ...lesson };
+    const copyConclusion = [...copyLesson.conclusion] as Array<any>;
+    copyConclusion[index] = con;
+    setLesson({ ...lesson, conclusion: copyConclusion });
```

Comment:
Possilby all these immutable copy/edits can be simplified with `Array.slice`, e.g.

```typescript
setLesson({ 
  ...lesson, 
  conclusion: [
    ...lesson.conclusion.slice(0, index), 
    con, 
    ...lesson.conclusion.slice(index + 1)
  ]
});
```

?